client/js/socket-events/join.js

```diff
@@ -10,11 +10,14 @@ const sidebar = $("#sidebar");
 socket.on("join", function(data) {
 	const id = data.network;
 	const network = sidebar.find(`#network-${id}-chanlist`);
-	network.append(
-		templates.chan({
-			channels: [data.chan],
-		})
-	);
+
+	const channels = network.children();
+	const position = $(channels[data.index || channels.length - 1]); // Put channel in correct position, or the end if we don't have one
+	const sidebarEntry = templates.chan({
+		channels: [data.chan],
+	});
+	$(sidebarEntry).insertBefore(position);
```

Comment:
Sanity check: can you confirm `position` (or `channels.length`) above could never be `0`?

Reply:
Well, the position is not a number itself, that's probably misleading name (but is taken from elsewhere). It's the actual element that we are placing above, if that makes sense. 

So I guess what you actually want is a check that data.index is not 0 or greater than channels.length - 1. On that point... I need to see what happens if we have no channels. This should probably break.

Reply:
So I completely forgot that we will always have a channel because the lobby is a channel, so it's fine. So it's not possible for there to be no channels.

This should be fine.